drivers/bridge/setup_ip_tables.go

```diff
@@ -13,14 +13,43 @@ const (
 	DockerChain = "DOCKER"
 )
 
-func (n *bridgeNetwork) setupIPTables(config *networkConfiguration, i *bridgeInterface) error {
+func setupIPChains(config *configuration) (*iptables.ChainInfo, *iptables.ChainInfo, error) {
 	// Sanity check.
 	if config.EnableIPTables == false {
-		return IPTableCfgError(config.BridgeName)
+		return nil, nil, fmt.Errorf("Cannot create new chains, driver EnableIPTable is disabled")
 	}
 
 	hairpinMode := !config.EnableUserlandProxy
 
+	natchain, err := iptables.NewChain(DockerChain, iptables.Nat)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to create NAT chain: %s", err.Error())
+	}
+
+	// TODO(banix): Move the following block; keep this function for just creating the chains
+	natchain, err = iptables.ProgramChain(natchain, hairpinMode)
+	if err != nil {
+		return natchain, nil, fmt.Errorf("Failed to program NAT chain: %s", err.Error())
+	}
+
+	filterchain, err := iptables.NewChain(DockerChain, iptables.Filter)
+	if err != nil {
+		return natchain, nil, fmt.Errorf("Failed to create FILTER chain: %s", err.Error())
```

Comment:
It should be possible for this function to revert the changes in case of failure.
After each step add  a defer function which would revert the step in case err != nil.
If revert fails, log a warning.